app/src/main/java/at/bitfire/icsdroid/ui/SubscriptionSettingsFragment.kt

```diff
@@ -5,157 +5,206 @@
-                model.defaultAlarmMinutes
-            )
-        )
-        model.defaultAllDayAlarmMinutes.observe(
-            viewLifecycleOwner,
-            defaultAlarmObserver(
-                binding.defaultAlarmAllDaySwitch,
-                binding.defaultAlarmAllDayText,
-                model.defaultAllDayAlarmMinutes
-            )
-        )
-
         val colorPickerContract = registerForActivityResult(ColorPickerActivity.Contract()) { color ->
             model.color.value = color
         }
-        binding.color.setOnClickListener {
-            colorPickerContract.launch(model.color.value)
+        return ComposeView(requireActivity()).apply {
+            setContent {
+                SubscriptionSettingsComposable(
+                    url = model.url.observeAsState("").value,
+                    title = model.title.observeAsState("").value,
+                    titleChanged = { model.title.postValue(it) },
+                    color = model.color.observeAsState(0).value,
+                    colorIconClicked = { colorPickerContract.launch(model.color.value) },
+                    ignoreAlerts = model.ignoreAlerts.observeAsState(false).value,
+                    ignoreAlertsChanged = { model.ignoreAlerts.postValue(it) },
+                    defaultAlarmMinutes = (model.defaultAlarmMinutes.observeAsState().value ?: "").toString(),
+                    defaultAlarmMinutesChanged = { model.defaultAlarmMinutes.postValue(stringToLongOrNull(it)) },
+                    defaultAllDayAlarmMinutes = (model.defaultAllDayAlarmMinutes.observeAsState().value ?: "").toString(),
+                    defaultAllDayAlarmMinutesChanged = { model.defaultAllDayAlarmMinutes.postValue(stringToLongOrNull(it)) }
+                )
+            }
         }
-
-        return binding.root
     }
 
-    /**
-     * Provides an observer for the default alarm fields.
-     * @param switch The switch view that updates the currently stored minutes.
-     * @param textView The viewer for the current value of the stored minutes.
-     * @param selectedMinutes The LiveData instance that holds the currently selected amount of minutes.
-     */
-    private fun defaultAlarmObserver(
-        switch: SwitchMaterial,
-        textView: TextView,
-        selectedMinutes: MutableLiveData<Long>
-    ) = Observer { min: Long? ->
-        switch.isChecked = min != null
-        // We add the listener once the switch has an initial value
-        switch.setOnCheckedChangeListener(getOnCheckedChangeListener(switch, selectedMinutes))
-
-        if (min == null)
-            textView.text = getString(R.string.add_calendar_alarms_default_none)
-        else {
-            val alarmPeriodText = PeriodFormat.wordBased().print(Minutes.minutes(min.toInt()))
-            textView.text = getString(R.string.add_calendar_alarms_default_description, alarmPeriodText)
-        }
+    private fun stringToLongOrNull(input: String) = try {
```

Comment:
Why not using [`toLongOrNull`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.text/to-long-or-null.html)? 🤔

Reply:
Oh, true haha.